tests/core/channels/test_audiocodes.py

```diff
@@ -0,0 +1,137 @@
+import pytest
+from rasa.core import utils
+
+
+def handle_activities_test_cb(expectedString):
+    async def cb(resultUM):
+        assert expectedString == resultUM.text
+
+    return cb
+
+
+def handle_activities_test_throw_cb(expectedString):
+    def cb(resultUM):
+        assert expectedString != resultUM.text
+
+    return cb
+
+
+def test_audiocodes_channel():
+
+    from rasa.core.channels.audiocodes import AudiocodesInput
+    import rasa.core
+
+    input_channel = AudiocodesInput("some_generated_token")
+
+    s = rasa.core.run.configure_app([input_channel], port=5004)
+
+    routes_list = utils.list_routes(s)
+    assert routes_list.get("ac_webhook.health", "").startswith("/webhooks/audiocodes")
+    assert routes_list.get("ac_webhook.receive", "").startswith(
+        "/webhooks/audiocodes/webhook"
+    )
+
+
+@pytest.mark.asyncio
+async def test_audiocodes_handle_events():
+
+    from rasa.core.channels.audiocodes import AudiocodesInput, AudiocodesOutput
+
+    ac_output = AudiocodesOutput()
+    conversation_id = "some-generated-uuid"
+
+    async def test_message(activity, expected_text, cb=handle_activities_test_cb):
+        message = {"conversation": conversation_id, "activities": [activity]}
+        return await c.handleActivities(message, ac_output, cb(expected_text))
+
+    input_channel = AudiocodesInput("some_generated_token")
+    c = AudiocodesInput.Conversation(conversation_id)
+    input_channel.conversations[conversation_id] = c
+
+    TEXT_MESSAGE = {
+        "type": "message",
+        "text": "This is a text message",
+        "parameters": {"confidence": 0.8172},
+        "id": "generated id 1",
+    }
+
+    await test_message(TEXT_MESSAGE, "This is a text message")
+
+    await test_message(
+        TEXT_MESSAGE, "This is a text message", cb=handle_activities_test_throw_cb
+    )
```

Comment:
Should be after the assertions. Your test fails because of that.

Reply:
It fails because of missing await in output_channel.send_custom_json